### ISSUE TITLE
Clicking scene instantiated via @tool script crashes editor

### ISSUE THREAD

**Issue by TAGames** [godotengine/godot#96116]
### Tested versions

v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 SUPER (NVIDIA; 32.0.15.6081) - AMD Ryzen 7 3800X 8-Core Processor (16 Threads)

### Issue description

Clicking on the object in editor instantiated with following script crashes the editor.

Console output:
Couldn't see if there was anything else printed above the same 27 lines:
```
[1] error(-1): no debug info in PE/COFF executable
.....
[27] error(-1): no debug info in PE/COFF executable
-- END OF BACKTRACE --
```

### Steps to reproduce

Script:
```
@tool
extends Node

const BLOCK = preload("res://block.tscn")

func _ready() -> void:
	var block = BLOCK.instantiate()
	add_child(block)
```

Block scene:

<img width="181" alt="image" src="https://github.com/user-attachments/assets/4f4713de-38f6-40ca-86d7-09d9fef2bac6">

### Minimal reproduction project (MRP)

[ToolCrashTest 26.08.2024 14.47.38.zip](https://github.com/user-attachments/files/16748734/ToolCrashTest.26.08.2024.14.47.38.zip)

Open test scene and click on the cube.

**Comment by yahkr** [comment]
This appears to be fixed in v4.3.1.rc.custom_build [ff9bc0422] & v4.4.dev.custom_build [804454240].  
Can you confirm?

**Comment by akien-mga** [comment]
> This appears to be fixed in v4.3.1.rc.custom_build [[ff9bc04](https://github.com/godotengine/godot/commit/ff9bc0422349219b337b015643544a0454d4a7ee)] & v4.4.dev.custom_build [804454240].

4.3.1.rc ff9bc04 is basically the same as 4.3.stable, nothing has been cherry-picked yet (the only difference is the version number). So if it's reproducible in official builds of 4.3.stable, and not in custom builds, it means the compiler is the main difference (official builds are made with mingw-gcc).

**Comment by CreatedBySeb** [comment]
It looks like the problem is a null pointer being passed to `Node3DEditorViewport::_is_node_locked` as a result of the changed code in `Node3DEditorViewport::_select_clicked` in the case that there is no editable parent of the clicked Node3D:

https://github.com/godotengine/godot/blob/142d33211cb4276dee6f797117981af983471b9e/editor/plugins/node_3d_editor_plugin.cpp#L741-L773

From debugging the provided MRP, it appears that the new code on lines 753-760 that walks up the scene tree to find the nearest selectable parent sets `selected` to null in the case that there is no selectable parent (such as in the case with the tool script). This null pointer is then passed into `_is_node_locked` on line 773, which does not check if the passed pointer is null and attempts to call `get_meta` on the null pointer, causing the bad memory access and crash.

To avoid a crash, there should probably be a null pointer check on the `selected` pointer, either inside of `_is_node_locked` (not sure what the correct default would be), or before attempting to check if the node is locked in `_select_clicked`, replacing line 773 with: 

```c++
if (p_allow_locked || (selected != nullptr && !_is_node_locked(selected))) {
```

similar to the check on line 755. Making this change solves the problem on macos.arm64, with the click having no effect rather than crashing the editor, which I believe is the intended behaviour. If this is the correct approach I can put that small change in a PR.
